packages/cosmic-swingset/lib/ag-solo/add-chain.js

```diff
@@ -22,11 +24,46 @@ async function addChain(basedir, chainConfig, force = false) {
     }
   }
 
-  console.log('downloading netconfig from', actualConfig);
-  const r = await fetch(actualConfig);
-  const netconf = await r.json();
+  const url = new URL(actualConfig, `file://${process.cwd()}`);
+  console.log('downloading netconfig from', url.href);
+  let netconf;
+  if (url.protocol === 'file:') {
+    const f = fs.readFileSync(url.pathname, 'utf-8');
+    netconf = JSON.parse(f);
+  } else {
+    const r = await fetch(url.href);
+    netconf = await r.json();
+  }
```

Comment:
I haven't dug deeply into this, but can't `fetch` read a `file://` URL?

Reply:
> I haven't dug deeply into this, but can't `fetch` read a `file://` URL?

Not any of the implementations I've seen (at least not this one).